homework/index.js

```diff
@@ -29,23 +31,141 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function addRow(table, title, repoText) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('th', tr, {
+      text: title,
+    });
+    createAndAppend('td', tr, { text: repoText });
+    return tr;
+  }
+
+  function renderRepoDetails(repo, parent) {
+    const date = new Date(repo.updated_at);
+    const table = createAndAppend('table', parent);
+    const firstRow = addRow(table, 'Repository:', '');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: '_blank',
+    });
+
+    addRow(table, 'Description:', repo.description, 'td');
+    addRow(table, 'Forks:', repo.forks, 'td');
+    addRow(table, 'Updated:', date.toLocaleDateString(), 'td');
+  }
+
+  function createOption(repo, parent, index) {
+    createAndAppend('option', parent, {
+      text: repo.name,
+      value: index,
+    });
+  }
+
+  function selectedIndex(selection, repositories, parent) {
```

Comment:
I think you misunderstood my previous comment with regard to the name of this function. It was first parameter for which I suggested to use the name `selectedIndex`, not the function name itself. Function names should usually indicate 'action', e.g. have an active verb in the name.

But in fact, you do not need this function at all. See my other comment on that below.